packages/editor-ui/src/components/helpers.ts

```diff
@@ -67,3 +67,7 @@ export function isString(value: unknown): value is string {
 export function isNumber(value: unknown): value is number {
 	return typeof value === 'number';
 }
+
+export function stringSizeInBytes(input: string): number {
+	return encodeURI(input).split(/%(?:u[0-9A-F]{2})?[0-9A-F]{2}|./).length - 1;
+}
```

Comment:
Looks a bit cryptic.

Reply:
UTF-16 strings can have characters like `%uXX` and `%uXXXX`. 

Essentially this has a lookahead for `%u` + a pair of optional hexadecimal characters `([0-9A-F]{2})?` + another `[0-9A-F]{2}` pair of hexadecimal values, or anything else `|.`.